File: common/py/openai.py
from openai import AsyncOpenAI
from agentuity import AgentRequest, AgentResponse, AgentContext

client = AsyncOpenAI()

def welcome():  
    return {  
        "welcome": "Welcome to the OpenAI Agent! I can help you interact with OpenAI models.",  
        "prompts": [  
            {  
                "data": "Generate a creative story about a robot learning to paint",  
                "contentType": "text/plain"  
            },  
            {  
                "data": "What capabilities does GPT-4 have?",  
                "contentType": "text/plain"  
            }  
        ]  
    }  

async def run(request: AgentRequest, response: AgentResponse, context: AgentContext):
    chat_completion = await client.chat.completions.create(
        messages=[
            {
                "role": "system",
                "content": "You are a friendly assistant!",
            },
            {
                "role": "user",
                "content": await request.data.text() or "Why is the sky blue?",
            },
        ],
        model="gpt-4o",
    )
    return response.text(chat_completion.choices[0].message.content)

File: common/js/openai.ts
import type { AgentRequest, AgentResponse, AgentContext } from "@agentuity/sdk";
import OpenAI from "openai";

const client = new OpenAI();

export const welcome = () => {  
	return {  
	  welcome: "Welcome to the OpenAI TypeScript Agent! I can help you interact with OpenAI models.",  
	  prompts: [  
		{  
		  data: "Generate a creative story about space exploration",  
		  contentType: "text/plain",  
		},  
		{  
		  data: "How can I use the OpenAI API for text completion?",  
		  contentType: "text/plain",  
		}  
	  ],  
	};  
  };

export default async function Agent(
	req: AgentRequest,
	resp: AgentResponse,
	ctx: AgentContext,
) {
	const completion = await client.chat.completions.create({
		messages: [
			{
				role: "user",
				content: (await req.data.text()) ?? "Say this is a test",
			},
		],
		model: "gpt-4o",
	});
	const message = completion.choices[0]?.message;
	return resp.text(message?.content ?? "Something went wrong");
}
